ament_cpplint/ament_cpplint/main.py

```diff
@@ -148,7 +148,9 @@ def main(argv=sys.argv[1:]):
         else:
             print("Not using '--root'")
         print('')
-        arguments += [f for f in files if any(excl not in f for excl in args.exclude)]
+        arguments_dirs = [f for f in files if os.path.dirname(f) not in args.excludes]
```

Comment:
This logic still doesn't cover the case where `--excludes` contain `foo` and a file `foo/bar/baz.h` should be skipped.

Reply:
Improved logic. https://github.com/ament/ament_lint/pull/238/commits/63d09b5f12884877d114de0d1ef8120d8f745815 
I added a condition If you decided to exclude a complete folder cpplnt should not execute this block of code. because cpplint is being called  by folders. First include, then src, etc... Sumarizing the script will return a failured test if there is/are no file/files to check without this condition

Reply:
friendly ping @dirk-thomas . Running CI for tf2 test avoiding `Linearmath` on cpplint, cppcheck and uncrustify https://github.com/ros2/geometry2/pull/258#issuecomment-636096343